src/components/DataDownloadModal/index.jsx

```diff
@@ -0,0 +1,228 @@
+        ).toFixed(2) + ' MB';
+        const labelSuffix = sampledataRegistry.Models[modelName].labelSuffix || '';
+        DatasetCheckboxRows.push(
+          <tr key={modelName}>
+            <td>
+              <Form.Check
+                className="pt-1"
+                id={modelName}
+              >
+                <Form.Check.Input
+                  type="checkbox"
+                  checked={dataListCheckBoxes[modelName]}
+                  onChange={(event) => this.handleCheckList(
+                    event, modelName
+                  )}
+                />
+                <Form.Check.Label>
+                  {modelName}
+                </Form.Check.Label>
+              </Form.Check>
+            </td>
+            <td><em>{labelSuffix}</em></td>
+            <td>{filesize}</td>
+          </tr>
+        );
+      });
+
+    return (
+      <Modal className="download-data-modal"
+        show={this.props.show}
+        onHide={this.props.closeModal}
+        size="lg"
+      >
+        <Form>
+          <Modal.Header>
+            <Modal.Title>Download InVEST sample data</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Table
+              size="sm"
+              borderless
+              striped
+            >
+              <thead>
+                <tr>
+                  <th>
+                    <Form.Check
+                      type="checkbox"
+                      id="all-sampledata"
+                      checked={this.state.allDataCheck}
+                      onChange={this.handleCheckAll}
+                      name="all-sampledata"
+                      label="Select All"
+                    />
+                  </th>
+                </tr>
+              </thead>
+              <tbody className="table-body">
+                {DatasetCheckboxRows}
+              </tbody>
+            </Table>
+          </Modal.Body>
+          <Modal.Footer>
+            <Button
+              variant="secondary"
+              onClick={this.props.closeModal}
+            >
+              Cancel
+            </Button>
+            <Button
+              variant="primary"
+              onClick={this.handleSubmit}
+              disabled={!downloadEnabled}
+            >
+              Download
+            </Button>
+          </Modal.Footer>
+        </Form>
+      </Modal>
+    );
+  }
+}
+
+DataDownloadModal.propTypes = {
+  show: PropTypes.bool.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  storeDownloadDir: PropTypes.func.isRequired,
+};
+
+export function DownloadProgressBar(props) {
+  const [nComplete, nTotal] = props.downloadedNofN;
+  if (nComplete === nTotal) {
+    return (
+      <Expire delay={props.expireAfter}>
+        <Alert variant="success">Download Complete</Alert>
+      </Expire>
+    );
+  }
+  return (
+    <ProgressBar
+      animated
+      max={1}
+      now={nComplete / nTotal}
+      label={`Downloading ${nComplete + 1} of ${nTotal}`}
+    />
```

Comment:
I think the progress bar is fine and works well for downloading individual model data. However, when selecting all and downloading it just sits with 1/1 and the bar to the far left... Doesn't really provide a visual indication of progress. But, I don't really think it's a big deal.

Reply:
Right, it does look a bit janky. We could also track progress by bytes downloaded instead of number of files. But the problem is we actually download and then extract, and at least for me, extraction takes longer. So I wanted the progress bar to wait for both steps. Anyway maybe we can make this look a little cleaner.

Sidenote: the download actually starts right away when the button is clicked, even before the user selects the save location. So the download is often already done before the save location is picked! Not certain how that works, but pretty cool.

Reply:
Quick fix for this was `now={(nComplete + 1) / nTotal}`

Still not very useful as progress indicator for a single file, but at least it looks more professional.

Reply:
Yea, that looks a lot better.

>So the download is often already done before the save location is picked!

That is super cool.